common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -465,7 +470,11 @@ public static void copyMemory(long srcAddr, byte[] dst, int dstIndex, long lengt
      * consumer (one thread!).
      */
     public static <T> Queue<T> newMpscQueue() {
-        return new MpscLinkedQueue<T>();
+        if (hasUnsafe()) {
+            return JAVA_VERSION < 8 ? new MpscLinkedQueue7<T>() : new MpscLinkedQueue8<T>();
+        } else {
```

Comment:
@normanmaurer I answered with a question: Do android support Unsafe? isn't that method ever called by Android users? That's the reason for the else.

If the answer to those conclude that it is redundant then just:

`return JAVA_VERSION < 8 ? new MpscLinkedQueue7<T>() : new MpscLinkedQueue8<T>();`

But it has to be done by @johnou since he cherry-pick and squashed the commit.